### PR TITLE
fix(deploy-docs): fix latest condition

### PR DESCRIPTION
**Pull request by KeisukeShima**
Related https://github.com/autowarefoundation/autoware-github-actions/issues/96

Added a condition for `latest` to correct a malfunction in `pull_request_target`.

Signed-off-by: Keisuke Shima <19993104+KeisukeShima@users.noreply.github.com>

### CODE REVIEW

**Review by kenji-miyake — DISMISSED** [review]
Have you tested this?
Please add your test result or some reference links to the description.

**Comment by kenji-miyake** [comment]
Unfortunately, I think you can't test this in this PR because it's `pull_request_target`.

**Comment by KeisukeShima** [comment]
> Unfortunately, I think you can't test this in this PR because it's `pull_request_target`.

First, I tried to display the variables, but they are not displayed...
https://github.com/autowarefoundation/autoware-documentation/pull/55/commits/5ccfa6ca1177a281f6601b2a094e23c4af1b588a

**Comment by KeisukeShima** [comment]
> Unfortunately, I think you can't test this in this PR because it's `pull_request_target`.

So the script is executed before the change. Understood.
But at least the contents of the variable should be displayed, right?

**Comment by kenji-miyake** [comment]
@KeisukeShima Hmm? What do you mean?
(If necessary, you can write your comments both in Japanese and English.)

**Comment by KeisukeShima** [comment]
> @KeisukeShima Hmm? What do you mean? (If necessary, you can write your comments both in Japanese and English.)

So the script is executed before the change. Understood.
But at least the contents of the variable should be displayed, right?
変更前のスクリプトが実行されるということですね。理解しました。
しかし、少なくとも変数の内容が表示されてもいいはずですよね。

という内容です

**Comment by KeisukeShima** [comment]
Note: The `pull_request_target` event does not reflect script changes :innocent:

File: .github/workflows/deploy-docs.yaml [hunk 1]
```diff
@@ -38,4 +38,9 @@ jobs:
         uses: autowarefoundation/autoware-github-actions/deploy-docs@v1
         with:
           token: ${{ secrets.GITHUB_TOKEN }}
-          latest: ${{ github.ref_name == github.event.repository.default_branch }}
+          latest: ${{ github.event.workflow_run.event != 'pull_request_target' && github.ref_name == github.event.repository.default_branch }}
```
Comment (KeisukeShima) [inline .github/workflows/deploy-docs.yaml:41]:
```suggestion
          latest: ${{ github.event_name != 'pull_request_target' && github.ref_name == github.event.repository.default_branch }}
```

https://docs.github.com/en/actions/learn-github-actions/contexts#example-usage-of-the-github-context

**Comment by KeisukeShima** [comment]
image
https://github.com/KeisukeShima/autoware-documentation/runs/5536759468?check_suite_focus=true

Variable values are as expected!

**Review by kenji-miyake — APPROVED** [review]

**Review by kenji-miyake — APPROVED** [review]

**Pull request merged**